src/deluge/gui/views/performance_session_view.cpp

```diff
@@ -0,0 +1,1931 @@
+	if (fxPress[xDisplay].padPressHeld) {
+		//obtain Param::Kind and ParamID corresponding to the column in focus (xDisplay)
+		Param::Kind lastSelectedParamKind = layoutForPerformance[xDisplay].paramKind; //kind;
+		int32_t lastSelectedParamID = layoutForPerformance[xDisplay].paramID;
+
+		if (lastSelectedParamID != kNoSelection) {
+			padReleaseAction(modelStack, lastSelectedParamKind, lastSelectedParamID, xDisplay, false);
+		}
+	}
+	updateLayoutChangeStatus();
+}
+
+/// check if parameter is stutter
+bool PerformanceSessionView::isParamStutter(Param::Kind paramKind, int32_t paramID) {
+	if ((paramKind == Param::Kind::UNPATCHED_SOUND) && (paramID == Param::Unpatched::STUTTER_RATE)) {
+		return true;
+	}
+	return false;
+}
+
+/// check if stutter is active and release it if it is
+void PerformanceSessionView::releaseStutter(ModelStackWithThreeMainThings* modelStack) {
+	if (isUIModeActive(UI_MODE_STUTTERING)) {
+		padReleaseAction(modelStack, Param::Kind::UNPATCHED_SOUND, Param::Unpatched::STUTTER_RATE,
+		                 lastPadPress.xDisplay, false);
+	}
+}
+
+/// this will set a new value for a parameter
+/// if we're dealing with stutter, it will check if stutter is active and end the stutter first
+/// if we're dealing with stutter, it will change the stutter rate value and then begin stutter
+/// if you're in the value editor, pressing a column and changing the value will also open the sound editor
+/// menu for the parameter to show you the current value in the menu
+/// in regular performance view, this function will also update the parameter value shown on the display
+bool PerformanceSessionView::setParameterValue(ModelStackWithThreeMainThings* modelStack, Param::Kind paramKind,
+                                               int32_t paramID, int32_t xDisplay, int32_t knobPos, bool renderDisplay) {
+	ModelStackWithAutoParam* modelStackWithParam = getModelStackWithParam(modelStack, paramID);
+
+	if (modelStackWithParam && modelStackWithParam->autoParam) {
+
+		if (modelStackWithParam->getTimelineCounter()
+		    == view.activeModControllableModelStack.getTimelineCounterAllowNull()) {
+
+			//if switching to a new pad in the stutter column and stuttering is already active
+			//e.g. it means a pad was held before, end previous stutter before starting stutter again
+			if ((paramKind == Param::Kind::UNPATCHED_SOUND) && (paramID == Param::Unpatched::STUTTER_RATE)
+			    && (isUIModeActive(UI_MODE_STUTTERING))) {
+				((ModControllableAudio*)view.activeModControllableModelStack.modControllable)
+				    ->endStutter((ParamManagerForTimeline*)view.activeModControllableModelStack.paramManager);
+			}
+
+			if (fxPress[xDisplay].previousKnobPosition == kNoSelection) {
+				int32_t oldParameterValue =
+				    modelStackWithParam->autoParam->getValuePossiblyAtPos(view.modPos, modelStackWithParam);
+				fxPress[xDisplay].previousKnobPosition =
```

Comment:
I think you should be storing the param value and not the knob pos - params have higher accuracy than knob position allows and this could cause issues in the future

Also you're converting the knob pos back to a param value before using it anyway

Reply:
Does this change need to be a part of this PR? As much as I want to make it, I don't know if I have the energy atm to go through and change everything from knobPos to value. I know it's not a complex change, but I do use the knobPos for the value displayed on the screen so I would want to make sure I'm not breaking things.